tools/ci_connector_ops/ci_connector_ops/pipelines/actions/builds.py

```diff
@@ -0,0 +1,81 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import tempfile
+from typing import List, Optional, Tuple
+
+import docker
+from ci_connector_ops.pipelines.utils import check_path_in_workdir
+from ci_connector_ops.utils import Connector
+from dagger.api.gen import Client, Container
+
+INSTALL_LOCAL_REQUIREMENTS_CMD = ["python", "-m", "pip", "install", "-r", "requirements.txt"]
+INSTALL_REQUIREMENTS_CMD = ["python", "-m", "pip", "install", "."]
+
+
+async def install(dagger_client: Client, connector_container: Container, extras: Optional[List] = None) -> Container:
+    """Create container in which the connector python package is installed with all its dependencies.
+
+    Args:
+        dagger_client (Client): _description_
+        connector_container (Container): _description_
+        extras (List, optional): List of extra_requires to intall e.g: tests, dev, main. (pip install -e .[tests]). Defaults to None.
+
+    Returns:
+        Container: A container in which the connector python package is installed with all its dependencies
+    """
+    if await check_path_in_workdir(connector_container, "requirements.txt"):
+        requirements_txt = await connector_container.file("requirements.txt").contents()
```

Comment:
Until https://github.com/dagger/dagger/issues/3764 is ready, you can still simplify the use case of needing a file's contents.

```python
async def get_file_contents(container: dagger.Container, path: str) -> str:
    try:
        return await container.file(path).contents()
    except dagger.QueryError as e:
        if "no such file or directory" not in str(e):
            # this is the hicky bit of the stopgap because 
            # this error could come from a network issue
            raise
    return None

async def install(dagger_client: Client, connector_container: Container, extras: Optional[List] = None) -> Container:
    if requirements_txt := await get_file_contents(connector_container, "requirements.txt"):
        ...
```

The advantage is it makes only one query.

Reply:
Thank you for the suggestion making me discover the walrus operator 🙏